### PR TITLE
TM-4016: aim individual error states - new

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**

### CODE REVIEW

**Comment by elizabeth-jimenez** [comment]
let's throw a margin top-top on our tm-spinner     
.tm-spinner {
      margin-top: 100px;
    }

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx [hunk 1]
```diff
@@ -31,6 +31,8 @@ const AgendaItemMaintenanceContainer = (props) => {
   const id = get(props, 'match.params.id'); // client's perdet
   const isCDO = get(props, 'isCDO');
   const client_data = useDataLoader(api().get, `/fsbid/client/${id}/`);
+  const clientDataLoading = get(client_data, 'data.loading') || '';
+  const clientDataError = get(client_data, 'data.error') || '';
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:35]: here we can use our shiny new format instead

```
const clientDataLoading = client_data?.data?.loading ?? '';
const clientDataError = client_data?.data?.error ?? '';
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 2]
```diff
@@ -58,10 +58,8 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
   const assignments = get(asgHistory, 'data') || [];
   const languages = get(clientData, 'data.data.languages') || [];
 
-  console.log('research pane remarks: ', remarks);
   const remarks_data = remarks?.data?.results?.filter(remark => remark.active_ind === 'Y') || [];
-  console.log('remarks_data: ', remarks_data);
-  const remarkCategories_data = get(remarkCategories, 'data.data.results') || [];
+  const remarkCategories_data = get(remarkCategories, 'data.results') || [];
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:62]: can use new format here too 😁

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx (hunk 1 above)
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:35]: `error` and `loading` are at the root of `client_data` and you'll want to default them to `false`

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 3]
```diff
@@ -161,6 +187,8 @@ AgendaItemResearchPane.propTypes = {
   ),
   legCount: PropTypes.number,
   isReadOnly: PropTypes.bool,
+  clientLoading: PropTypes.string,
+  clientError: PropTypes.string,
 };
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:192]: should be Booleans

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 4]
```diff
@@ -37,27 +37,34 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
...
-    isReadOnly } = props;
+    isReadOnly, clientLoading, clientError } = props;
 
   const [selectedNav, setSelectedNav] = useState(get(tabs, '[0].value') || '');
   const classifications = useSelector(state => state.classifications);
+  const classificationsError = useSelector(state => state.classificationsHasErrored);
+  const classificationsLoading = useSelector(state => state.classificationsIsLoading);
   const clientClassifications = useSelector(state => state.userClassifications);
+  const clientClassificationsLoading = useSelector(state => state.userClassificationsIsLoading);
+  const clientClassificationsError = useSelector(state => state.userClassificationsHasErrored);
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:48]: 🙌

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 5]
```diff
@@ -37,27 +37,34 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
...
-  // eslint-disable-next-line no-unused-vars
+
+  const { data: asgHistory, error: asgHistError, loading: asgHistLoading } = useDataLoader(api().get, `/fsbid/assignment_history/${perdet}/`);
+  const { data: remarks, error: remarksDataError, loading: remarksDataLoading } = useDataLoader(api().get, '/fsbid/agenda/remarks/');
   const { data: frequentPositionsResults, error: frequentPositionsError, loading: frequentPositionsLoading } = useDataLoader(api().get, '/fsbid/positions/frequent_positions/');
-  const remarkCategories = useDataLoader(api().get, '/fsbid/agenda/remark-categories/');
+  const { data: remarkCategories, error: rmrkCatError, loading: rmrkCatLoading } = useDataLoader(api().get, '/fsbid/agenda/remark-categories/');
 
-  const assignments = get(data, 'data') || [];
+  const assignments = get(asgHistory, 'data') || [];
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:58]: here too for new format

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 6]
```diff
@@ -37,27 +37,34 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
...
 
-  const { data, error, loading } = useDataLoader(api().get, `/fsbid/assignment_history/${perdet}/`);
-  const { data: remarks } = useDataLoader(api().get, '/fsbid/agenda/remarks/');
-  // eslint-disable-next-line no-unused-vars
+
+  const { data: asgHistory, error: asgHistError, loading: asgHistLoading } = useDataLoader(api().get, `/fsbid/assignment_history/${perdet}/`);
+  const { data: remarks, error: remarksDataError, loading: remarksDataLoading } = useDataLoader(api().get, '/fsbid/agenda/remarks/');
   const { data: frequentPositionsResults, error: frequentPositionsError, loading: frequentPositionsLoading } = useDataLoader(api().get, '/fsbid/positions/frequent_positions/');
-  const remarkCategories = useDataLoader(api().get, '/fsbid/agenda/remark-categories/');
+  const { data: remarkCategories, error: rmrkCatError, loading: rmrkCatLoading } = useDataLoader(api().get, '/fsbid/agenda/remark-categories/');
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:56]: let's either update everywhere we have `remarks` to `rmrk` or vice-versa

Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:56]: for the variable names, i mean

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 2 above)
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:62]: can also rename to `remarkCategories$`

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 7]
```diff
@@ -79,6 +86,64 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
...
+          return errorAlert;
+        }
+        return (<FrequentPositions
+          positions={frequentPositions}
+          addFrequentPosition={addFrequentPosition}
+          disabled={((legCount >= 10) || isReadOnly)}
+        />);
+
+      case TP:
+        if (clientClassificationsError || classificationsError) {
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:124]: 🙌

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx [hunk 8]
```diff
@@ -79,6 +86,64 @@ const AgendaItemResearchPane = forwardRef((props = { perdet: '', clientData: {},
...
+          return errorAlert;
+        }
+        return (<div id="aim-classifications"> {/* needed for css specificity */}
+          <Classifications {...classificationsProps} />
+        </div>);
+
+      case RG:
+        if (remarksDataError || rmrkCatError) {
```
Comment (elizabeth-jimenez) [inline src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:132]: 🦜

File: src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx (hunk 1 above)
Comment (pveland621) [reply src/Components/Agenda/AgendaItemMaintenanceContainer/AgendaItemMaintenanceContainer.jsx:35]: done!

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 2 above)
Comment (pveland621) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:62]: done!

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 5 above)
Comment (pveland621) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:58]: done!

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 3 above)
Comment (pveland621) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:192]: done!

File: src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx (hunk 6 above)
Comment (pveland621) [reply src/Components/Agenda/AgendaItemResearchPane/AgendaItemResearchPane.jsx:56]: done!

**Review by pveland621 — APPROVED** [review]
whoever coded this must be a genius

**Pull request merged**